src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,190 @@
+     *
+     * @return Either Done or Delete Command
+     *
+     * @throws DukeException If the format is wrong or if a non-integer is passed in as argument or  multiple arguments
+     *                       are passed in
+     */
+    private Command parseDoneDelete(String input) throws DukeException {
+        StringTokenizer st = new StringTokenizer(input);
+        // if there are more than 2 tokens, then it's wrong format:
+        if (st.countTokens() != 2) {
+            throw new DukeException(Message.ERROR_DONEDELETE_ARGS.getMsg());
+        }
+        String command = st.nextToken();
+        String taskID = st.nextToken();
+        if (!isInteger(taskID)) {
+            throw new DukeException(Message.ERROR_DONEDELETE_NOTINT.getMsg());
+        }
+        String[] parsedInput = new String[]{command, taskID};
+        return command.equals(CommandWord.DELETE_CMD.getCmd())
+               ? new DeleteCommand(parsedInput)
+               : new DoneCommand(parsedInput);
+    }
+    /**
+     * Parses user input that is related to the todo
+     *
+     * @param input User input related to todo command
+     *
+     * @return String[] array of format: description of todo
+     *
+     * @throws DukeException If no description is passed in
+     */
+    private String[] parseToDo(String input) throws DukeException {
+        StringTokenizer st = new StringTokenizer(input);
+        st.nextToken();
+        if (!st.hasMoreTokens()) {
+            throw new DukeException(Message.ERROR_TODO_DESC.getMsg());
+        }
+        StringBuilder description = new StringBuilder();
+        while (st.hasMoreTokens()) {
+            description.append(st.nextToken()).append(" ");
+        }
+        return new String[]{"T", description.toString().stripTrailing()};
+    }
+    /**
+     * Parses user input that is related to the Deadline command
+     *
+     * @param input User input related to Deadline command
+     *
+     * @return String array of format: description of Deadline String representation for the date
+     *
+     * @throws DukeException If the format is wrong or if it lacks proper description
+     */
+    private String[] parseDeadline(String input) throws DukeException {
+        String[] separatedInput = input.split(DEADLINE_DELIMITER);
+        if (separatedInput.length <= 1) {
+            throw new DukeException(Message.ERROR_DEADLINE_FORMAT.getMsg());
+        }
+        String dateString = separatedInput[1];
+        StringTokenizer st = new StringTokenizer(separatedInput[0]);
+        st.nextToken();
+        StringBuilder descriptionBuilder = new StringBuilder();
+        while (st.hasMoreTokens()) {
+            descriptionBuilder.append(st.nextToken()).append(" ");
+        }
+        String description = descriptionBuilder.toString().stripTrailing();
+        if (description.isEmpty()) {
+            throw new DukeException(Message.ERROR_DEADLINE_DESC.getMsg());
+        }
+        return new String[]{"D",
```

Comment:
Maybe can considering wrapping these three lines into one, 'cause it does not seem complicated to me :)

Reply:
Hi, I'm not sure what you mean though. Are you referring to the following part:

```java
        String description = descriptionBuilder.toString().stripTrailing();
        if (description.isEmpty()) {
            throw new DukeException(Message.ERROR_DEADLINE_DESC.getMsg());
        }
```

I think it's better to make the predicate for if statements as intuitive as possible, hence the `description` var.